nilearn/masking.py

```diff
@@ -607,18 +612,28 @@ def compute_brain_mask(target_img, threshold=.5, connected=True,
         Controls the amount of verbosity: higher numbers give
         more messages
 
+    mask_type : {'whole-brain', 'gm', 'wm'}, optional
+        Type of mask to be computed. Default = 'whole-brain'
+
     Returns
     -------
     mask : nibabel.Nifti1Image
         The whole-brain mask (3D image)
     """
     if verbose > 0:
-        print("Template mask computation")
+        msg_to_print = "Template " + mask_type + " mask computation"
+        print(msg_to_print)
 
     target_img = _utils.check_niimg(target_img)
 
-    from .datasets import load_mni152_brain_mask
-    template = load_mni152_brain_mask()
+    if mask_type == 'whole-brain':
+        template = load_mni152_brain_mask()
+    elif mask_type == 'gm':
+        template = fetch_icbm152_brain_gm_mask()
+    else:
+        assert mask_type == 'wm'
+        template = fetch_icbm152_brain_wm_mask()
```

Comment:
Not sure, but shouldn't we update the `warning_msg` argument of `_post_process_mask` below depending on this?